RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -0,0 +1,202 @@
+    L1TJetsMatching<T>::~L1TJetsMatching(){ }
+    
+    template< typename T>
+    void L1TJetsMatching<T>::produce(edm::StreamID iSId, edm::Event& iEvent, const edm::EventSetup& iES) const
+    {
+        
+        unique_ptr<std::vector<T>> pfMatchedJets(new std::vector<T>);
+        std::pair<std::vector<T>,std::vector<T>> output;
+        
+        double deltaR    = 1.0;
+        double matchingR = 0.5;
+        
+        // Getting HLT jets to be matched
+        edm::Handle<std::vector<T> > pfJets;
+        iEvent.getByToken( jetSrc_, pfJets );
+        
+        edm::Handle<trigger::TriggerFilterObjectWithRefs> l1TriggeredJets;
+        iEvent.getByToken(jetTrigger_,l1TriggeredJets);
+        
+        //l1t::TauVectorRef jetCandRefVec;
+        l1t::JetVectorRef jetCandRefVec;
+        l1TriggeredJets->getObjects( trigger::TriggerL1Jet,jetCandRefVec);
+        
+        math::XYZPoint a(0.,0.,0.);
+        
+        //std::cout<<"PFsize= "<<pfJets->size()<<endl<<" L1size= "<<jetCandRefVec.size()<<std::endl;
+        for(unsigned int iJet = 0; iJet < pfJets->size(); iJet++){
+            for(unsigned int iL1Jet = 0; iL1Jet < jetCandRefVec.size(); iL1Jet++){
+                // Find the relative L2pfJets, to see if it has been reconstructed
+                const T &  myJet = (*pfJets)[iJet];
+                //  if ((iJet<3) && (iL1Jet==0))  std::cout<<myJet.p4().Pt()<<" ";
+                deltaR = ROOT::Math::VectorUtil::DeltaR2(myJet.p4().Vect(), (jetCandRefVec[iL1Jet]->p4()).Vect());
+                if(deltaR < matchingR ) {
+                    pfMatchedJets->push_back(myJet);
+                    break;
+                }
+            }
+        }
+        
+        output= categorise(*pfMatchedJets,pt1_Min,pt2_Min, mjj_Min);
+        unique_ptr<std::vector<T>> output1(new std::vector<T>(output.first));
+        unique_ptr<std::vector<T>> output2(new std::vector<T>(output.second));
+        
+        iEvent.put(std::move(output1),"TwoJets");
+        iEvent.put(std::move(output2),"ThreeJets");
+        
+    }
+    template< typename T>
+     void L1TJetsMatching<T>::fillDescriptions(edm::ConfigurationDescriptions& descriptions)
+     {
+     edm::ParameterSetDescription desc;
+     desc.add<edm::InputTag>("L1JetTrigger", edm::InputTag("hltL1DiJetVBF"))->setComment("Name of trigger filter"    );
+     desc.add<edm::InputTag>("JetSrc"      , edm::InputTag("hltAK4PFJetsTightIDCorrected"))->setComment("Input collection of PFJets");
+     desc.add<double>       ("pt1_Min",95.0)->setComment("Minimal pT1 of PFJets to match");
+     desc.add<double>       ("pt2_Min",35.0)->setComment("Minimal pT2 of PFJets to match");
+     desc.add<double>       ("mjj_Min",650.0)->setComment("Minimal mjj of matched PFjets");
+     descriptions.setComment("This module produces collection of PFJetss matched to L1 Taus / Jets passing a HLT filter (Only p4 and vertex of returned PFJetss are set).");
+     descriptions.add       ("L1TJetsMatching",desc);
```

Comment:
Needs to be done differently for templated classes as otherwise all instances generate a cfi with the same name.
something like:
```
#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
descriptions.add(defaultModuleLabel<L1TJetsMatching<T>>(), desc);
```